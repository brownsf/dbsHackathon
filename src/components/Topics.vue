<template>
  <v-layout row>
    <v-flex
      sm12
      md8
      offset-md2>
      <v-card>
        <v-toolbar
          color="teal"
          dark>
          <v-toolbar-title>Topics</v-toolbar-title>
          <v-spacer/>
          <v-btn
            icon
            @click.stop="addDialog = true">
            <v-icon>add</v-icon>
          </v-btn>
        </v-toolbar>
        <v-progress-linear
          v-if="loading"
          :indeterminate="true"/>
        <v-list
          two-line
          subheader>
          <v-list-tile
            v-for="topic in topics"
            :key="topic.id">
            <v-list-tile-content>
              <v-list-tile-title>{{ topic.name }}</v-list-tile-title>
              <v-list-tile-sub-title>{{ topic.description }}</v-list-tile-sub-title>
            </v-list-tile-content>
            <v-list-tile-action @click="vote(topic.id)">
              <v-badge
                left
                color="red">
                <span slot="badge">{{ topic.votes }}</span>
                <v-icon>thumb_up</v-icon>
              </v-badge>
            </v-list-tile-action>
          </v-list-tile>
        </v-list>
      </v-card>
    </v-flex>
    <v-dialog
      v-model="addDialog"
      max-width="500px">
      <v-card>
        <v-card-title>
          Add Topic
        </v-card-title>
        <v-card-text>
          <v-layout row>
            <v-flex
              xs12
              sm8
              offset-sm2>
              <v-text-field
                v-model="name"
                :rules="[v=>!!v||'Name is required']"
                label="Name"
                validate-on-blur
              />
            </v-flex>
          </v-layout>
          <v-layout row>
            <v-flex
              xs12
              sm8
              offset-sm2>
              <v-text-field
                :rules="descRules"
                v-model="description"
                label="Description"
                validate-on-blur
              />
            </v-flex>
          </v-layout>
        </v-card-text>
        <v-card-actions>
          <v-btn
            color="primary"
            flat
            @click="saveTopic">Save</v-btn>

          <v-btn
            color="secondary"
            flat
            @click.stop="addDialog=false">Cancel</v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
     <v-dialog
        v-model="dialog"
        fullscreen
        hide-overlay
        transition="dialog-bottom-transition"
        scrollable
      >
      comments go here
     </v-dialog>
  </v-layout>

</template>
<script>
import { mapState } from 'vuex';

export default {
  data: () => ({
    addDialog: false,
    description: '',
    name: '',
    descRules: [
      v => !!v || 'Description is required',
      v => v.length <= 25 || 'Max 25 characters',
    ],
  }),

  computed: mapState({
    topics: state => state.topics,
    loading: state => state.topicLoad,
  }),
  created() {
    if (this.$store.getters.isAuth) {
      this.$store.dispatch('getAllTopics');
    } else {
      this.$router.push('/login');
    }
  },
  methods: {
    saveTopic() {
      this.$store
        .dispatch('addTopic', {
          name: this.name,
          description: this.description,
        })
        .then(() => {
          this.addDialog = false;
          this.$store.dispatch('getAllTopics');
        });
    },
    vote(id) {
      this.$store.dispatch('vote', { id }).then(() => {});
    },
  },
};
</script>
