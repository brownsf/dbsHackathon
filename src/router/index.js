import Vue from 'vue';
import Router from 'vue-router';
import Register from '@/components/Register';
import Login from '@/components/Login';
import Topics from '@/components/Topics';
import Main from '@/components/Main';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Main',
      component: Main,
    },
    {
      path: '/register',
      name: 'Register',
      component: Register,
    },
    {
      path: '/login',
      name: 'Login',
      component: Login,
    },
    {
      path: '/topics',
      name: 'Topics',
      component: Topics,
    },
  ],
});
