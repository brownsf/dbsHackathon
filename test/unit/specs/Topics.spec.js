import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Topics from '@/components/Topics';
import router from '@/router';

Vue.use(Vuex);
Vue.use(Vuetify);
Vue.use(VueRouter);
describe('Comments', () => {
  const state = { data: { topic: { name: 'here' }, singleTopicLoad: false } };
  const actions = {
    register: jest.fn(),
    getAllTopics: jest.fn(),
    login: jest.fn(),
    getTopic: jest.fn(),
    addComment: jest.fn(),
    addTopic: jest.fn(),
  };
  const getters = {
    isAuth: () => true,
    singleTopic: () => ({
      comments: [
        {
          message: 'one',
        },
        { message: 'two' },
      ],
    }),
  };

  const store = new Vuex.Store({
    state,
    actions,
    getters,
  });
  const wrapper = mount(Topics, { store, router });
  it('should render', () => {
    // expect(wrp.html()).toMatchSnapshot();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should save comment', () => {
    wrapper.vm.$refs.form.validate = () => true;
    wrapper.find('.saveTopic').trigger('click');
    expect(actions.addTopic.mock.calls).toHaveLength(1);
  });

  it('should render state corrently', () => {
    expect(wrapper.findAll('.loader')).toHaveLength(0);
    expect(wrapper.findAll('.errorAlert')).toHaveLength(0);
  });
});
