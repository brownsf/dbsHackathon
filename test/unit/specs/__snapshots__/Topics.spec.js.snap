// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should redirect 1`] = `undefined`;

exports[`Comments should render 1`] = `
<div class="layout row">
  <div class="flex sm12 md8 offset-md2">
    <div class="card" style="height: auto;">
      <nav class="toolbar theme--dark teal" style="margin-top: 0px; padding-right: 0px; padding-left: 0px;">
        <div class="toolbar__content" style="height: 56px;">
          <div class="toolbar__title">Topics</div>
          <div class="spacer"></div>
          <button type="button" class="btn btn--icon">
            <div class="btn__content">
              <i aria-hidden="true" class="icon material-icons">add</i>
            </div>
          </button>
        </div>
      </nav>
      <!---->
      <div class="list list--subheader list--two-line">
        <div class="topicItem">
          <div>
            <div class="list__tile">
              <div class="list__tile__content">
                <div class="list__tile__title">yep</div>
                <div class="list__tile__sub-title"></div>
              </div>
              <div class="list__tile__action voter"><span class="badge badge--left"> <i aria-hidden="true" class="icon material-icons">thumb_up</i><span class="badge__badge red"><span></span></span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="dialog__container" style="display: block;">
    <div class="dialog__content" tabindex="-1" style="z-index: 0;">
      <div class="dialog" style="max-width: 500px; display: none;">
        <div class="card" style="height: auto;">
          <form novalidate="novalidate">
            <div class="card__title">
              Add Topic
            </div>
            <div class="card__text">
              <div class="layout row">
                <div class="flex xs12 sm8 offset-sm2">
                  <div class="input-group input-group--text-field primary--text">
                    <label>Name</label>
                    <div class="input-group__input">
                      <input tabindex="0" aria-label="Name" type="text">
                    </div>
                    <div class="input-group__details">
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
              <div class="layout row">
                <div class="flex xs12 sm8 offset-sm2">
                  <div class="input-group input-group--text-field primary--text">
                    <label>Description</label>
                    <div class="input-group__input">
                      <input tabindex="0" aria-label="Description" type="text">
                    </div>
                    <div class="input-group__details">
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="card__actions">
              <button type="button" class="saveTopic btn btn--flat primary--text">
                <div class="btn__content">Save</div>
              </button>
              <button type="button" class="btn btn--flat secondary--text">
                <div class="btn__content">Cancel</div>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div class="dialog__container" style="display: block;">
    <div class="dialog__content" tabindex="-1" style="z-index: 0;">
      <div class="dialog dialog--fullscreen dialog--scrollable" style="display: none;">
        <div class="commentContainer">
          <div class="card card--tile" style="height: auto;">
            <nav class="toolbar toolbar--card theme--dark primary" style="margin-top: 0px; padding-right: 0px; padding-left: 0px;">
              <div class="toolbar__content" style="height: 56px;">
                <button type="button" class="btn btn--icon theme--dark">
                  <div class="btn__content">
                    <i aria-hidden="true" class="icon material-icons">close</i>
                  </div>
                </button>
                <div class="toolbar__title">Comments for </div>
                <div class="spacer"></div>
                <div class="toolbar__items">
                  <button type="button" class="btn btn--flat theme--dark">
                    <div class="btn__content">add</div>
                  </button>
                </div>
              </div>
            </nav>
            <div class="container grid-list-md text-xs-center">
              <div class="layout row wrap">
                <div class="flex xs12">
                  <div class="list list--subheader">
                    <div class="subheader">Recent comments:</div>
                    <!---->
                    <!---->
                    <div>
                      <div class="list__tile">
                        <div class="list__tile__content">
                          <div class="list__tile__title">one</div>
                        </div>
                      </div>
                    </div>
                    <hr class="divider">
                    <div>
                      <div class="list__tile">
                        <div class="list__tile__content">
                          <div class="list__tile__title">two</div>
                        </div>
                      </div>
                    </div>
                    <hr class="divider">
                  </div>
                </div>
              </div>
            </div>
            <div class="dialog__container" style="display: block;">
              <div class="dialog__content" tabindex="-1" style="z-index: 0;">
                <div class="dialog" style="max-width: 500px; display: none;">
                  <div class="card" style="height: auto;">
                    <div class="card__title">
                      Add Comment
                    </div>
                    <form novalidate="novalidate">
                      <div class="card__text">
                        <div class="layout row">
                          <div class="flex xs12 sm8 offset-sm2">
                            <div class="input-group input-group--text-field input-group--multi-line primary--text">
                              <label>Comment</label>
                              <div class="input-group__input">
                                <textarea tabindex="0" aria-label="Comment" rows="5"></textarea>
                              </div>
                              <div class="input-group__details">
                                <!---->
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div class="card__actions">
                        <button type="button" class="saveComment btn btn--flat primary--text">
                          <div class="btn__content">Add</div>
                        </button>
                        <button type="button" class="btn btn--flat secondary--text">
                          <div class="btn__content">Cancel</div>
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
